diff --git a/node_modules/@convex-dev/better-auth/.bun-tag-23870342cb6792e7 b/.bun-tag-23870342cb6792e7
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@convex-dev/better-auth/.bun-tag-96ff0bb2d1bc5aba b/.bun-tag-96ff0bb2d1bc5aba
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/dist/plugins/convex/index.js b/dist/plugins/convex/index.js
index 030e7f18b940ee5ddecea125800a6c7d6ab52ff3..bef131cef7d6b49372b4bcab154f0de085985008 100644
--- a/dist/plugins/convex/index.js
+++ b/dist/plugins/convex/index.js
@@ -178,13 +178,13 @@ export const convex = (opts) => {
                 },
                 {
                     matcher: (ctx) => {
-                        console.log(ctx.path, ctx.context.session);
+                        
                         return (ctx.path?.startsWith("/sign-out") ||
                             ctx.path?.startsWith("/delete-user") ||
                             (ctx.path?.startsWith("/get-session") && !ctx.context.session));
                     },
                     handler: createAuthMiddleware(async (ctx) => {
-                        console.log("clearing jwt cookie");
+                        
                         const jwtCookie = ctx.context.createAuthCookie(JWT_COOKIE_NAME, {
                             maxAge: 0,
                         });
diff --git a/dist/plugins/cross-domain/client.js b/dist/plugins/cross-domain/client.js
index ea276eac573ab935faf76e9af4e98a9f9192acc0..8772ce33b8ab91446268a7a289bf12d878814b8c 100644
--- a/dist/plugins/cross-domain/client.js
+++ b/dist/plugins/cross-domain/client.js
@@ -131,7 +131,7 @@ export const crossDomainClient = (opts = {}) => {
                             const prevCookie = await storage.getItem(cookieName);
                             const toSetCookie = getSetCookie(setCookie || "", prevCookie ?? undefined);
                             await storage.setItem(cookieName, toSetCookie);
-                            store?.notify("$sessionSignal");
+                            const isGetSession = context.request.url.toString().includes("/get-session"); if (!isGetSession) { store?.notify("$sessionSignal"); }
                         }
                         if (context.request.url.toString().includes("/get-session") &&
                             !opts?.disableCache) {
diff --git a/src/plugins/cross-domain/client.ts b/src/plugins/cross-domain/client.ts
index 40f9f95d82f0d3478dc79b6d4d6f694a5ee8ecdf..4b00a37d7ed8b36084942d6b82ac9aa33952649b 100644
--- a/src/plugins/cross-domain/client.ts
+++ b/src/plugins/cross-domain/client.ts
@@ -174,7 +174,10 @@ export const crossDomainClient = (
                 prevCookie ?? undefined
               );
               await storage.setItem(cookieName, toSetCookie);
-              store?.notify("$sessionSignal");
+              const isGetSession = context.request.url.toString().includes("/get-session"); 
+              if (!isGetSession) { 
+                store?.notify("$sessionSignal"); 
+              }
             }
 
             if (
