diff --git a/dist/index.edge.d.ts b/dist/index.edge.d.ts
index d5256be2f92ad1b1fbb4b7b163025b9a750b5b0e..302bb8d7e6c6131b2e172a1d96f91e56f964a92d 100644
--- a/dist/index.edge.d.ts
+++ b/dist/index.edge.d.ts
@@ -1 +1,5 @@
-export declare const registerOTel: (serviceName: string) => void;
+export declare const registerOTel: (
+  serviceName: string,
+  url: string,
+  headers: object
+) => void;
diff --git a/dist/index.node.d.ts b/dist/index.node.d.ts
index d5256be2f92ad1b1fbb4b7b163025b9a750b5b0e..302bb8d7e6c6131b2e172a1d96f91e56f964a92d 100644
--- a/dist/index.node.d.ts
+++ b/dist/index.node.d.ts
@@ -1 +1,5 @@
-export declare const registerOTel: (serviceName: string) => void;
+export declare const registerOTel: (
+  serviceName: string,
+  url: string,
+  headers: object
+) => void;
diff --git a/dist/index.node.js b/dist/index.node.js
index 5fb3fe21827daecfe3344f33aa1c24fc3704c939..c340dbe0a3a0c3a2cc93626db3c7c76d3887a484 100644
--- a/dist/index.node.js
+++ b/dist/index.node.js
@@ -3,13 +3,18 @@ import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-grpc";
 import { Resource } from "@opentelemetry/resources";
 import { SemanticResourceAttributes } from "@opentelemetry/semantic-conventions";
 import { SimpleSpanProcessor } from "@opentelemetry/sdk-trace-node";
-export const registerOTel = (serviceName) => {
-    const sdk = new NodeSDK({
-        resource: new Resource({
-            [SemanticResourceAttributes.SERVICE_NAME]: serviceName,
-        }),
-        spanProcessor: new SimpleSpanProcessor(new OTLPTraceExporter()),
-    });
-    sdk.start();
+export const registerOTel = (serviceName, url, headers) => {
+  const sdk = new NodeSDK({
+    resource: new Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: serviceName,
+    }),
+    spanProcessor: new SimpleSpanProcessor(
+      new OTLPTraceExporter({
+        url,
+        headers,
+      })
+    ),
+  });
+  sdk.start();
 };
 //# sourceMappingURL=index.node.js.map
