diff --git a/index.js b/index.js
index e480b5cf2285aae782298399c110d5a2bf816deb..2bb5662d0c1d4100ff4e2b6a24f32fc428fdd31c 100644
--- a/index.js
+++ b/index.js
@@ -1,4 +1,4 @@
-throw new Error(
-  "This module cannot be imported from a Client Component module. " +
-    "It should only be used from a Server Component."
-);
+// throw new Error(
+//   "This module cannot be imported from a Client Component module. " +
+//     "It should only be used from a Server Component."
+// );
